arch/arm/arm64/tls.c

```diff
@@ -59,11 +59,25 @@
 
 /*
  * Aarch64 ABI requires 16 bytes of space after the tlsp as part of the TCB.
- * NOTE: We assume that libraries (like musl) reserve this space when using
- *       this TLS API. You can find more information here:
+ * NOTE: Some libraries, such as older versions of musl, reserve this space
+ *       when using this TLS API. You can find more information here:
  *       https://www.openwall.com/lists/musl/2018/02/09/7
+ *
+ * Newer versions of musl, however, simply ignore this block of 16 bytes
+ * described by the ABI and expect the tlsp to point just at the end of
+ * the TCB. More details can be found here:
+ * https://www.openwall.com/lists/musl/2018/06/01/14
+ *
+ * Therefore, when such libraries are used, they can call the makefile rule
+ * `aarch64_no_reserved_tcb_overlap` which sets a macro if the library
+ * **does not** expect a 16 bytes reserved block as part of the TCB,
+ *  as is the case with newer versions of musl.
  */
+#ifdef CONFIG_AARCH64_NO_RESERVED_TCB_OVERLAP
+#define AARCH64_RESERVED 0
+#else
 #define AARCH64_RESERVED 16
+#endif
```

Comment:
Good point, makes sense. I have kept the `AARCH64_RESERVED` macro for the ABI reserved 16 byte block and introduced a new macro `AARCH64_TCB_OVERLAP` which is set to 0 when there is no TCB overlap. Otherwise, it is set to `AARCH64_RESERVED`.